src/Input.js

```diff
@@ -3,8 +3,32 @@ import styled from "styled-components";
 import PropTypes from "prop-types";
 
 const Input = ({ onChange, value, label, type }) => {
+
+  const calculateOffset = () => {
```

Comment:
Nice! This is super cool to see this approach. I bet we can get it pretty close.

I would extract this function out from the component into a stand-alone function. I.e. move the definition down the the bottom of the file and have the function take a `value` parameter. We could test then (a) write tests for this function and (2) explicitly show that it's a "pure" function that doesn't rely on any outside logic or values. See `src/Results.js` where the functions to calculate each result value are outside of the component definition.

Reply:
Ok. Yes. I see.